components/core/src/clp/FileReader.cpp

```diff
@@ -12,15 +12,27 @@
 using std::string;
 
 namespace clp {
+FileReader::FileReader(string const& path)
+        : m_file{fopen(path.c_str(), "rb")},
+          m_getdelim_buf_len(0),
+          m_getdelim_buf(nullptr) {
+    if (nullptr == m_file) {
+        if (ENOENT == errno) {
+            throw OperationFailed(ErrorCode_FileNotFound, __FILE__, __LINE__);
+        }
+        throw OperationFailed(ErrorCode_errno, __FILE__, __LINE__);
+    }
+    m_path = path;
+}
+
 FileReader::~FileReader() {
-    close();
+    // NOTE: We don't check errors for fclose since it seems the only reason it could fail is if
+    // it was interrupted by a signal
+    fclose(m_file);
```

Comment:
Should we check if m_file != nullptr? technically it should never happen though.